parquet/src/encryption/decrypt.rs

```diff
@@ -149,6 +149,25 @@ impl FileDecryptionProperties {
     pub fn builder(footer_key: Vec<u8>) -> DecryptionPropertiesBuilder {
         DecryptionPropertiesBuilder::new(footer_key)
     }
+
+    /// Retrieval of key used for encryption of footer and (possibly) columns
+    pub fn footer_key(&self) -> &Vec<u8> { self.footer_key.as_ref() }
+
+    /// Get the column names, keys used in column_keys
+    pub fn column_keys(&self) -> (Vec<String>, Vec<Vec<u8>>) {
```

Comment:
This will be incompatible with my key retriever changes: https://github.com/apache/arrow-rs/pull/7286

We could make it return an option though?

I guess exposing the `fn column_key(&self, column_name: &str, key_metadata: Option<&[u8]>) -> Result<Cow<Vec<u8>>>` method isn't enough for you as it doesn't tell you which columns are encrypted? Although we don't know that if a key retriever is used.

Reply:
Shouldn't particularly be a problem with your new PR. I expect this could just return empty vectors in the case of a key retriever + add a getter for the key retriever. The main point is to have getters so that FileEncryptionProperties can be recreated.